gateway/hostname.go

```diff
@@ -121,7 +83,7 @@ func WithHostname(next http.Handler, api IPFSBackend, publicGateways map[string]
 			if !gw.NoDNSLink && hasDNSLinkRecord(r.Context(), api, host) {
 				// rewrite path and handle as DNSLink
 				r.URL.Path = "/ipns/" + stripPort(host) + r.URL.Path
-				next.ServeHTTP(w, withHostnameContext(r, host))
+				next.ServeHTTP(w, withSubdomainContext(r, host))
```

Comment:
:thought_balloon:  why are we using `withSubdomainContext`  ad not `withDNSLinkContext` here?

iiuc this is  a code path that  handles DNSLink in `Host` header (not `dweb.link/ipns/example.com` but `example.com` case)

Reply:
We never used the equivalent of `withDNSLinkContext` here. I don't know. Probably because the code is too flaky and if we did set it, then the cross-origin check would return `true` and links would be incorrect. I never wrote this code, but I think that's the reason.

Reply:
We never used the equivalent of `withDNSLinkContext` here. I'm not really sure why. I will add it and check if sharness passes. It looks like a bug but I don't really see why there aren't complaints. It's possible it was something that was missed on https://github.com/ipfs/kubo/pull/8890